test/fast/Unit tests/nvm_print_color_code

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+set -ex
+
+die () { echo "$@" ; exit 1; }
+
+\. ../../../nvm.sh
+
+# Testing valid input
+OUTPUT=$(nvm_print_color_code m)
+EXPECTED_OUTPUT='0;35m'
+[ "_$OUTPUT" = "_$EXPECTED_OUTPUT" ] || die "nvm_print_color_code returned wrong code; got $(echo ">$OUTPUT<") expected $(echo ">$EXPECTED_OUTPUT<")"
+
+# Testing invalid input
+set +ex # needed for stderr
+OUTPUT="$(nvm_print_color_code q 2>&1)"
+set -ex
```

Comment:
```suggestion
set +x # needed for stderr
OUTPUT="$(nvm_print_color_code q 2>&1)"
set -x
```
does this work?

Reply:
Hi @ljharb, we tested it with just `set +x` and `set -x`, but it does not work. So we do need the `e`.

Reply:
hmm - that suggests that it wouldn't work in production without `set -e`, no?

Reply:
Hmm, this needs some further discussion.

Reply:
Turns out that this is because `nvm_print_color_code` is failing, by necessity. One option is to keep the `+e`, another would be:
```suggestion
set +x # needed for stderr
OUTPUT="$(nvm_print_color_code q 2>&1)" ||:
set -x
```
(or possibly `OUTPUT="$(nvm_print_color_code q 2>&1 ||:)"`)

I'm content with either approach.

Reply:
Hi @ljharb, this did work with the `||:` at the end of the line 🥳. Thanks for your help with that.